Add documentation to Rcpp interfaces

Bai-Li-NOAA opened the pull request:
<!---
Thanks for opening a PR. This commented text will **NOT** appear in the final PR. Toggle between Write and Preview to see what your PR will look like without the comments.
-->

# What is the feature?
* This PR addresses issue #635. Documentation for Rcpp interfaces has been added to the codebase. Users can view the documentation in the R console by using the FIMS::: prefix. For example,

```r
> FIMS:::AgeComp
C++ class 'AgeComp' <00000171049c3560>
Constructors:
    AgeComp(int, int)
        docstring : The first dimension of the data is the maximum year, and the second dimension is the maximum age.

Fields: 
    Rcpp::Vector<14, Rcpp::PreserveStorage> age_comp_data
        docstring : The age composition data.

Methods: 
     unsigned int get_id()  
           docstring : Get the ID of the interface object, No inputs are required.
```
# How have you implemented the solution?
* Added documentation to `inst/include/interface/rcpp/rcpp_interface.hpp`
* Fixed minor documentation typos in `inst/include/interface/rcpp/rcpp_objects/rcpp_data.hpp` and `inst/include/interface/rcpp/rcpp_objects/rcpp_population.hpp`

# Does the PR impact any other area of the project, maybe another repo?
* No.

kellijohnson-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
```diff
@@ -318,151 +318,246 @@ void clear() {
-      .field("log_rzero", &BevertonHoltRecruitmentInterface::log_rzero)
-      .field("log_devs", &BevertonHoltRecruitmentInterface::log_devs)
+      .field("logit_steep", &BevertonHoltRecruitmentInterface::logit_steep,
+             "Recruitment relative to unfished recruitment at 20% of unfished spawning biomass. Should be a value between 0.2 and 1.0.")
```
It might be good to spell out percent rather than using the sign just in case there are downstream effects, for example within latex it would need to be escaped.

Bai-Li-NOAA replied:
Done.

kellijohnson-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
If this is the natural log instead of "log" it would be nice to be more specific. And, the line is missing a full stop at the end.

Andrea-Havron-NOAA replied:
I prefer ln_ instead of log_ for this reason.

Bai-Li-NOAA replied:
@Andrea-Havron-NOAA, this means we need to refactor several names in multiple places (e.g., recruitment_base.hpp, sr_beverton_holt.hpp, rcpp_recruitment.hpp, rcpp_population.hpp for recruitment-related code). Is it okay for me to change all relevant parameter names from log_ to ln_ in this pull request?

kellijohnson-NOAA replied:
We are going to be doing an overhaul of parameter names (by @peterkuriyama-NOAA) soon that goes beyond log to ln so if you want @Bai-Li-NOAA you could just update the documentation and @peterkuriyama-NOAA will change the names later.

Bai-Li-NOAA replied:
Thanks @kellijohnson-NOAA, I will just update the documentation in this pull request!

Bai-Li-NOAA replied:
Done.

kellijohnson-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
Missing full stop.

kellijohnson-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
Missing full stop.

kellijohnson-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
Missing full stop here and line 401.

kellijohnson-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
Missing full stop.

kellijohnson-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
Change the comma to a full stop.

kellijohnson-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
Change the comma to a full stop.

kellijohnson-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
Change stock to population.

kellijohnson-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
"The" should be "the".

kellijohnson-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
This line and line 549 are missing full stops at the end.

Andrea-Havron-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
```diff
@@ -318,151 +318,246 @@ void clear() {
+      .field("value", &Parameter::value_m,
+             "Initial value of the parameter.")
+      .field("min", &Parameter::min_m,
+             "Min value of the parameter.")
```
Default value is -Infinity

Andrea-Havron-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
```diff
@@ -318,151 +318,246 @@ void clear() {
+      .field("min", &Parameter::min_m,
+             "Min value of the parameter.")
+      .field("max", &Parameter::max_m,
+             "Max value of the parameter.")
```
Default value is Infinity

Andrea-Havron-NOAA commented on inst/include/interface/rcpp/rcpp_interface.hpp:
'or' a survey

Bai-Li-NOAA commented:
Thank you, @kellijohnson-NOAA and @Andrea-Havron-NOAA, for your thorough review. I have addressed all of your suggestions. Please let me know if you have any additional feedback. The PR still requires approval from one of you.

kellijohnson-NOAA commented:
Thank you @Bai-Li-NOAA for all of this documentation. I needed it in dev not main so I unfortunately had to redo the edits in cleanup_dev. The commit message gives you attribution, I hope that is sufficient and I am sorry that I was not able to just merge this branch in.
